tests/nlu/parsers/test_integer.py

```diff
@@ -0,0 +1,33 @@
+"""
+This module contains the tests for the integer parser
+"""
+from spokestack.nlu.parsers import integer
+
+
+def test_integer():
+    metadata = {"range": [1, 50]}
+    raw_value = "one"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 1
+
+    raw_value = "1"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 1
+
+    raw_value = "word"
+    parsed = integer.parse(metadata, raw_value)
+    assert not parsed
+
+    raw_value = "oneth"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 1
+
+    metadata = {"range": [1, 500]}
+    raw_value = "four hundred four"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 404
+
+    metadata = {"range": [1, 500]}
+    raw_value = "four hundred hundred"
+    parsed = integer.parse(metadata, raw_value)
+    assert not parsed
```

Comment:
It's sort of unclear just from looking at the test whether this fails because it's out of range or because it's an ill-formed number. You'll also want to exercise different parts of the map—exponents, teens, multiples of 10 + other numbers, etc.